### DISCUSSION TITLE
How do I query for this data with Prisma?

### DISCUSSION THREAD

**Discussion by mikecfisher** [prisma/prisma#3682] [Q&A] [answered]
I'm working on a feature and trying to use Prisma. I know enough sql to do it but Prisma is a bit new to me. I've read the docs and watched some videos on the Prisma YouTube channel but this still isn't immediately clear to me. 

What I need to do is: 

"Write a function that takes a date and returns a list of locations set to get reports on this date (say Monday) and the users for each location that have reporting enabled" 

So something like 

```js
{
  "location-1": {
     users: [{name: 'user-1'}, {name:'user-2'}]
  },
  "location-2":{
      users:  [{name: 'user-3'}, {name: 'user-4}]
  }
}
```
I'm told I can get this data via regular sql by doing a join from 3 tables `location > user_location > user` 

Here's the abridged `prisma.schema` for these tables

```prisma
model location {
  uid                          String                  @id
  reporting_date      DateTime?
  user_location         user_location[]
  @@index([company_uid], name: "locations_by_company_uid")
  @@index([settings_package_uid], name: "locations_by_settings_package_uid")
}

model user_location {
  location_uid String
  user_uid       String
  location        location  @relation(fields: [location_uid], references: [uid])
  user             user      @relation(fields: [user_uid], references: [uid])

  @@id([location_uid, user_uid])
  @@index([location_uid], name: "user_locations_by_location_uid")
  @@index([user_uid], name: "user_locations_by_user_uid")
}

model user {
  uid                           String          @id
  email                       String
  name                       String?
  reporting_enabled Boolean?
  user_location           user_location[]
}
```

Any help would be really appreciated. I know how to query for basic data via Prisma but this more complicated query isn't immediately clear after reading the docs.

**Comment by ryands17** [comment, marked as answer]
Hey @mikefisher84 👋 

As per your schema, I think the following should work:

```ts
  const data = await prisma.location.findMany({
    where: { reporting_date: date },
    select: {
      user_location: {
        select: { user: true, location: true },
        where: { user: { reporting_enabled: true } },
      },
    },
  })
```

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
